docs/sdg/sdg-refactor.md

```diff
@@ -0,0 +1,136 @@
+- The ezxpected RAG functionality in 2025 will have some complex interactions with both preprocessing and postprocessing, perhaps even involving user flows in which the core SDG functionality is not needed.  In that case, it would be confusing to have the code path for RAG include a call out to the SDG library for doing preprocessing but not actually doing the core SDG.
+- It would just be simpler to explain to all stakeholders if the functionality that I've been calling "core SDG" was really just called "SDG".  We can't do that now because the SDG library has preprocessing and postprocessing in it too.
+
+Conclusion:
+
+- While the cons here are substantial, so are the pros.  None of the cons really seem disqualifying.  The first pro (future changes to the formats can be more self-contained) seems particularly compelling because this is a rapidly evolving field and adding new expressive power seems like something we will want frequently.
+
+### Option 2: Move preprocessing and postprocessing into a new repository
+
+We could have a new repository for preprocessing and postprocessing and move all the preprocessing and postprocessing code there.
+
+Pro:
+
+- Avoids all the cons of Option 1.
+- Preprocessing and postprocessing are a coherent pieces of functionality that *could* have their own library (or libraries, FWIW).
+
+Cons:
+
+- Avoids all the pros of Option 1.
+- Having a separate repository with its own library brings in an enormous amount of overhead in maintaining that repository (e.g., CI/CD).
+- Having a separate repository with its own library also brings in an enormous amount of overhead in maintaining the CLI repository's dependency on all of those libraries.
+- Does not allow user flow 2.1 but maybe that's OK because it is not a priority and anyway the users could approximate that flow by also installing the ingestion library.
+
+Conclusion:
+
+- The cost of having a separate reposity is so high that we would only consider this option as a last resort.
+
+### Option 3: Move preprocessing and postprocessing into the CLI repository
+
+Pro:
+
+- The CLI already has a lot of "supporting" (non-core) functionality, so it would respect established precedent to include preprocessing and/or postprocessing here.
```

Comment:
I think a key `pro` left out here is the amount of functionality in the CLI:

1. download
2. serve
3. chat
4. train (community)
5. list
6. edit
7. init

it contains most user facing logic aside from what we call the "core" parts of the workflow (SDG, Train, Eval). Moving taxonomy/doc processing to the CLI fits with the design intention of the 3 libraries when they were created.

Reply:
I guess I would argue that this is an additional level of detail to the pro in the line above, not a missing pro.   I will expand on the existing bullet and add sub-bullets rather than add more top-level bullets for these points.